src/iter/collect/consumer.rs

```diff
@@ -31,7 +23,9 @@ impl<'c, T: Send + 'c> CollectConsumer<'c, T> {
 /// the elements will be dropped, unless its ownership is released before then.
 #[must_use]
 pub(super) struct CollectResult<'c, T> {
-    start: *mut T,
+    /// A slice covering the target memory, initialized up to our separate `len`.
+    target: &'c mut [MaybeUninit<T>],
+    /// The current initialized length in `target`
     len: usize,
     invariant_lifetime: PhantomData<&'c mut &'c mut [T]>,
```

Comment:
you should be able to get rid of this.

Reply:
I *think* you're right, but I'm not certain why it wanted to be invariant in `'c` in the first place. Simple `&'c mut [T]` would have already been invariant in `T`. @bluss, you wrote that in #759, do you recall why?